realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -149,6 +152,13 @@ public boolean isValid() {
      */
     public RealmQuery<E> where() {
         realm.checkIfValid();
+        try {
```

Comment:
Why not throw `IllegalStateException` and say in the message that you are querying on data that was deleted? I don't see the problem with multiple ISE? 
Returning null is not a good option IMO.